src/components/item/settings/GeolocationPicker.tsx

```diff
@@ -0,0 +1,153 @@
+import { ChangeEventHandler, useCallback, useEffect, useState } from 'react';
+
+import Clear from '@mui/icons-material/Clear';
+import {
+  Box,
+  IconButton,
+  LinearProgress,
+  List,
+  ListItemButton,
+  Stack,
+  TextField,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { OpenStreetMapProvider } from 'leaflet-geosearch';
+
+import { useBuilderTranslation } from '@/config/i18n';
+import { hooks, mutations } from '@/config/queryClient';
+import { BUILDER } from '@/langs/constants';
+import useDebouncedCallback from '@/utils/useDebounce';
+
+type SearchResult = {
+  osm_id: string;
```

Comment:
Maybe a comment could help here, if you want this smaller name ?
```suggestion
  osm_id: string; // Open Street Map id
```